dc/s2n-quic-dc/src/packet/tag.rs

```diff
@@ -79,6 +79,7 @@ decoder_value!(
                     Ok((Self::UnknownPathSecret(tag), buffer))
                 }
                 // reserve this range for other packet types
+                #[allow(clippy::match_overlapping_arm)]
                 0b0110_0000..=0b0111_1111 => Err(s2n_codec::DecoderError::InvariantViolation(
                     "unexpected packet tag",
                 )),
```

Comment:
It didn't seem right to have to update this range whenever a new packet type was added, so I ignored the clippy warning. Let me know if it would be better to specify the range like this though:
```
0b0110_0011..=0b0111_1111
```